activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -727,29 +739,11 @@ def configure_connection
 
           super
 
-          # Enforce foreign key constraints
-          # https://www.sqlite.org/pragma.html#pragma_foreign_keys
-          # https://www.sqlite.org/foreignkeys.html
           raw_execute("PRAGMA foreign_keys = ON", "SCHEMA")
-          unless @memory_database
-            # Journal mode WAL allows for greater concurrency (many readers + one writer)
-            # https://www.sqlite.org/pragma.html#pragma_journal_mode
-            raw_execute("PRAGMA journal_mode = WAL", "SCHEMA")
-            # Set more relaxed level of database durability
-            # 2 = "FULL" (sync on every write), 1 = "NORMAL" (sync every 1000 written pages) and 0 = "NONE"
-            # https://www.sqlite.org/pragma.html#pragma_synchronous
-            raw_execute("PRAGMA synchronous = NORMAL", "SCHEMA")
-            # Set the global memory map so all processes can share some data
-            # https://www.sqlite.org/pragma.html#pragma_mmap_size
-            # https://www.sqlite.org/mmap.html
-            raw_execute("PRAGMA mmap_size = #{128.megabytes}", "SCHEMA")
+
+          (@config.keys & PRAGMA_STATEMENTS).each do |key|
+            @raw_connection.send "#{key}=", @config[key]
```

Comment:
I preferred using the setters offered by the sqlite gem rather than executing the PRAGMA statements directly because the [gem provides normalization](https://github.com/sparklemotion/sqlite3-ruby/blob/261103408a45bd63aed81aa3cce685cd8bf85fb6/lib/sqlite3/pragmas.rb) of the values before setting them.

Reply:
If I understand [this code](https://github.com/sparklemotion/sqlite3-ruby/blob/261103408a45bd63aed81aa3cce685cd8bf85fb6/lib/sqlite3/pragmas.rb#L65) correctly, it will also accept something like `journal_mode=(:wal)`. Which I think is nicer than `journal_mode: WAL` in the config.

Reply:
Yep, same could be done for for `synchronous`